### PR TITLE
Debugging warnings in pss workflow

### PR DESCRIPTION
**Pull request by biswajit-9776**
# Pull Request Template for Kubeflow manifests Issues

## ✅ Contributor checklist
  - Make sure you have tested with kustomize. See [Installation Prerequisites](https://github.com/kubeflow/manifests#prerequisites)
  - All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

---     
 
> You can join the CNCF Slack and access our meetings at the [Kubeflow Community](https://www.kubeflow.org/docs/about/community/) website. Our channel on the CNCF Slack is here [**#kubeflow-platform**](https://app.slack.com/client/T08PSQ7BQ/C073W572LA2).

### CODE REVIEW

**Comment by juliusvonkohout** [comment]
Loogs good, now you just need to create patches to fix

```
Run ./tests/gh-actions/enable_restricted_PSS.sh
Patching the PSS-restricted labels for namespace istio-system...
namespace/istio-system patched
Patching the PSS-restricted labels for namespace auth...
namespace/auth patched
Patching the PSS-restricted labels for namespace cert-manager...
namespace/cert-manager patched
Patching the PSS-restricted labels for namespace oauth2-proxy...
Warning: existing pods in namespace "oauth2-proxy" violate the new PodSecurity enforce level "restricted:latest"
Warning: oauth2-proxy-5bcc78bdc9-7f24j: allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
namespace/oauth2-proxy patched
Patching the PSS-restricted labels for namespace kubeflow...
Warning: existing pods in namespace "kubeflow" violate the new PodSecurity enforce level "restricted:latest"
namespace/kubeflow patched
Warning: cache-server-5dbfbbdc95-bk6rl (and 12 other pods): unrestricted capabilities, runAsNonRoot != true, runAsUser=0, seccompProfile
Warning: cache-server-64cbf5d9f6-6h8nh (and 9 other pods): allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, runAsUser=0, seccompProfile
Warning: kubeflow-pipelines-profile-controller-7b7b8f44f7-ktpxd (and 1 other pod): allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
Warning: metacontroller-0: seccompProfile
```

**Comment by biswajit-9776** [comment]
```shell
- name: Apply patches to clear warnings
      run: |
        DIRECTORY="contrib/security/PSS/patches"
        for file in "$DIRECTORY"/*.yaml; do
          echo "Patching file: $file"
          KIND=$(kubectl get -f "$file" -o jsonpath='{.kind}')
          NAME=$(kubectl get -f "$file" -o jsonpath='{.metadata.name}')
          NAMESPACE=$(kubectl get -f "$file" -o jsonpath='{.metadata.namespace}')
          # Apply the patch
          kubectl get "$KIND" "$NAME" -n "$NAMESPACE" &> /dev/null
          if [ $? -eq 0 ]; then
            kubectl patch "$KIND" "$NAME" -n "$NAMESPACE" --patch-file "$file"
          fi
        done
``` 
This job in our workflow was added to fix the warnings. Locally it does so but I don't see why it seems to not work here.
Let me restart some of the deployments to see if that works.

**Comment by juliusvonkohout** [comment]
I also see for example https://github.com/kubeflow/manifests/blob/da0255f10d875040c2d845cd61b7938236c0dfaa/contrib/security/PSS/patches/oauth2-proxy.yaml#L7 which should not be the case. It seems to patch the replicas.

**Comment by juliusvonkohout** [comment]
maybe you need to wait for all deployments to restart (pods) after the patches and become ready.

**Comment by biswajit-9776** [comment]
> I also see for example
> 
> https://github.com/kubeflow/manifests/blob/da0255f10d875040c2d845cd61b7938236c0dfaa/contrib/security/PSS/patches/oauth2-proxy.yaml#L7
> 
> which should not be the case. It seems to patch the replicas.

Ohh, this _replicas: 2_ is just to have 2 similar pods in the same namespace. However, this line isn't needed so I'll remove it

**Comment by biswajit-9776** [comment]
@juliusvonkohout the kubeflow pods fail to restart successfully, even before 10 minutes. Any suggestions?

**Comment by juliusvonkohout** [comment]
maybe some of them are failing due to the security context. maybe they are in a crashloopbackoff

**Comment by juliusvonkohout** [comment]
Maybe you should add the readiness wait  to your patch loop and test one by one this way

**Comment by biswajit-9776** [comment]
@juliusvonkohout Hey, how much wait time should I set for each pod? Apparently 180s isn't enough.

**Comment by juliusvonkohout** [comment]
> @juliusvonkohout Hey, how much wait time should I set for each pod? Apparently 180s isn't enough.

Either it is not enough or it is in a crashloop. try 5 minutes then and output the status of the pod at 1 minute intervals or so,

**Comment by juliusvonkohout** [comment]
Maybe some of them need uid 1000 gid 0

**Comment by biswajit-9776** [comment]
I was checking if we can have all the failing pods run with UID 1000 so I intentionally didn't add them to some pods to see if it works.
I will add them in subsequent commits.

**Comment by juliusvonkohout** [comment]
Yes, only add 1000/0 where needed

**Comment by biswajit-9776** [comment]
Has there recently been any changes to profile controller? They seem to be getting timed out at 300s

**Comment by juliusvonkohout** [comment]
Please try runasgroup: 0. This might fix the cache-server

**Comment by juliusvonkohout** [comment]
See also the PSP attachments in https://github.com/kubeflow/manifests/issues/2014

File: contrib/security/PSS/patches/kubeflow-pipelines-profile-controller.yaml [hunk 1]
```diff
@@ -13,6 +13,8 @@ spec:
           seccompProfile:
             type: RuntimeDefault
           runAsNonRoot: true
+          runAsGroup: 1000
```
[the remark this answers is not in the document]
Comment (juliusvonkohout) [reply contrib/security/PSS/patches/kubeflow-pipelines-profile-controller.yaml]: Check that in kustomize build or your live cluster.

[the remark this answers is not in the document]
Comment (juliusvonkohout) [reply contrib/security/PSS/patches/kubeflow-pipelines-profile-controller.yaml]: It is probably 2.2.0 or so.

**Comment by biswajit-9776** [comment]
```
shell
docker inspect kubeflownotebookswg/kfam
[
    {
        "Id": "sha256:ef0104d66a715db475b319524b127344bf9311dae840b123fd24e8428673af39",
        "RepoTags": [
            "kubeflownotebookswg/kfam:latest"
        ],
        "RepoDigests": [
            "kubeflownotebookswg/kfam@sha256:07280ecc6e558c9923c5770d9e39d848e65f6b00bea4d266355fddee8f500ad3"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2024-01-27T17:19:15.158937816Z",
        "Container": "",
        "ContainerConfig": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "0",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "8081/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/access-management"
            ],
            "ArgsEscaped": true,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "/",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 60982937,
        "VirtualSize": 60982937,
```
@juliusvonkohout In the Config.User: 0, I guess the kfam image runs as root.

**Comment by biswajit-9776** [comment]
The cache-server uses a non-numeric "User": "appuser", so I guess giving it a _runAsUser: 1000_ should work.

**Comment by juliusvonkohout** [comment]
User 0 is just the default. That does not mean that there is a particular reason. I got all of them running rootless with the PSP settings I sent you.

**Comment by biswajit-9776** [comment]
Yes, so shoulr _runAsUser: 1000_ would be okay or any other ID would be preferrable?
For securityContext it's usual to use 1000 as UID.
Should I also include GID:0?

**Comment by juliusvonkohout** [comment]
> Yes, so shoulr _runAsUser: 1000_ would be okay or any other ID would be preferrable?
> For securityContext it's usual to use 1000 as UID.
> Should I also include GID:0?

Yes, just start with UID 1000 and GID 0 as mentioned last week.

**Comment by biswajit-9776** [comment]
Hey there @juliusvonkohout, I have tried adding the UIDs and GIDs, and also viewed the deployments using describe command where they seem to be okay in status after pacthes are applied. Also I have tried reapplying the kubeflow resources again but they are getting timed out.

**Comment by juliusvonkohout** [comment]
Why do you even have the reapply step? You install once and patch once at the beginning. If possible you can even merge those steps. The patch should trigger a new replica set and pod for deployments. Then baseline PSS and restricted PSS.

**Comment by juliusvonkohout** [comment]
Keep it simple and straightforward.

**Comment by juliusvonkohout** [comment]
@lampajr can you help here?

**Comment by juliusvonkohout** [comment]
@MaxKavun can you help here as well?

**Comment by biswajit-9776** [comment]
Should I atleast have wait commands for the resources? Unfortunately that leads to timeout of those resources.

**Comment by lampajr** [comment]
> @lampajr can you help here?

Hey @juliusvonkohout , even if am not that familiar with these PSS test I am happy to help here.

@biswajit-9776 I am not sure what is actually failing here (I see tests are succeeding now). Is there any issue with `metadata-grpc-deployment`?

**Comment by biswajit-9776** [comment]
Hey there @lampajr, thanks for reaching out. I have been trying to patch the deployments of Pipelines with securityContext after when the PSS labels are applied to the kubeflow namespace, the warnings that appear in the _Apply PSS labels to namespaces_ workflow should not be present.
```shell
Run ./tests/gh-actions/enable_restricted_PSS.sh
Patching the PSS-restricted labels for namespace istio-system...
namespace/istio-system patched
Patching the PSS-restricted labels for namespace auth...
namespace/auth patched
Patching the PSS-restricted labels for namespace cert-manager...
namespace/cert-manager patched
Patching the PSS-restricted labels for namespace oauth2-proxy...
namespace/oauth2-proxy patched
Patching the PSS-restricted labels for namespace kubeflow...
namespace/kubeflow patched
Warning: existing pods in namespace "kubeflow" violate the new PodSecurity enforce level "restricted:latest"
Warning: cache-server-b9bcff9fd-rdx7g (and 12 other pods): unrestricted capabilities, runAsNonRoot != true, runAsUser=0, seccompProfile
Warning: metacontroller-0: seccompProfile
Warning: ml-pipeline-9f[4](https://github.com/kubeflow/manifests/actions/runs/10834707411/job/30064531887?pr=2866#step:10:5)fb47b6-r6t4l: allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, runAsUser=0, seccompProfile
``` 
We initially suspected that the pods after the patches are applied aren't up or healthy so we added kubectl wait commands for all the deployments of pipelines. Some pods like `metadata-grpc-deployment` failed the timeout.

**Comment by lampajr** [comment]
> Hey there @lampajr, thanks for reaching out. I have been trying to patch the deployments of Pipelines with securityContext after when the PSS labels are applied to the kubeflow namespace, the warnings that appear in the _Apply PSS labels to namespaces_ workflow should not be present.
> 
> ```shell
> Run ./tests/gh-actions/enable_restricted_PSS.sh
> Patching the PSS-restricted labels for namespace istio-system...
> namespace/istio-system patched
> Patching the PSS-restricted labels for namespace auth...
> namespace/auth patched
> Patching the PSS-restricted labels for namespace cert-manager...
> namespace/cert-manager patched
> Patching the PSS-restricted labels for namespace oauth2-proxy...
> namespace/oauth2-proxy patched
> Patching the PSS-restricted labels for namespace kubeflow...
> namespace/kubeflow patched
> Warning: existing pods in namespace "kubeflow" violate the new PodSecurity enforce level "restricted:latest"
> Warning: cache-server-b9bcff9fd-rdx7g (and 12 other pods): unrestricted capabilities, runAsNonRoot != true, runAsUser=0, seccompProfile
> Warning: metacontroller-0: seccompProfile
> Warning: ml-pipeline-9f[4](https://github.com/kubeflow/manifests/actions/runs/10834707411/job/30064531887?pr=2866#step:10:5)fb47b6-r6t4l: allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, runAsUser=0, seccompProfile
> ```

I might be wrong but I think the warning are there because those pods have some invalid `securityContextx.capabilities`, I tried locally to apply the `baseline` PSS and looking at the `cache-server` configuration I see:

```yaml
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        add:
        - NET_ADMIN
        - NET_RAW
        drop:
        - ALL
```

And in according to the [documentation](https://kubernetes.io/docs/concepts/security/pod-security-standards/), `NET_ADMIN` and `NET_RAW` are not among the allowed values for `capabilities.add`. I am not an expert so it is just a guess :) 

> We initially suspected that the pods after the patches are applied aren't up or healthy so we added kubectl wait commands for all the deployments of pipelines. Some pods like `metadata-grpc-deployment` failed the timeout.

This timeout failure might be because the metadata-grpc-deployment could take a looot of time to get ready as it needs to wait for mysql deployment first and then it can start up correctly (it took more than 3 mins in my local env so that might be the reason of the failure - not 100% sure though)

[1] https://kubernetes.io/docs/concepts/security/pod-security-standards/

**Comment by biswajit-9776** [comment]
I see, thanks @lampajr, I will increase the timeout then and find out what's wrong with cache-server meanwhile.

**Comment by lampajr** [comment]
> I see, thanks @lampajr, I will increase the timeout then and find out what's wrong with cache-server meanwhile.

Keep in mind that also other pods are affected (in kubeflow namespace) not just cache-server (even if they are not listed explicitly) but at least the issue looks the same (at least for the baseline) - not 100% sure that fixing those patches is the right way to go, maybe @juliusvonkohout has more insights on this

**Comment by juliusvonkohout** [comment]
With istio-cni we can drop net_admin and net_raw for the proxy sidecars i think. So the tests here should use it.

**Comment by biswajit-9776** [comment]
> > Hey there @lampajr, thanks for reaching out. I have been trying to patch the deployments of Pipelines with securityContext after when the PSS labels are applied to the kubeflow namespace, the warnings that appear in the _Apply PSS labels to namespaces_ workflow should not be present.
> > ```shell
> > Run ./tests/gh-actions/enable_restricted_PSS.sh
> > Patching the PSS-restricted labels for namespace istio-system...
> > namespace/istio-system patched
> > Patching the PSS-restricted labels for namespace auth...
> > namespace/auth patched
> > Patching the PSS-restricted labels for namespace cert-manager...
> > namespace/cert-manager patched
> > Patching the PSS-restricted labels for namespace oauth2-proxy...
> > namespace/oauth2-proxy patched
> > Patching the PSS-restricted labels for namespace kubeflow...
> > namespace/kubeflow patched
> > Warning: existing pods in namespace "kubeflow" violate the new PodSecurity enforce level "restricted:latest"
> > Warning: cache-server-b9bcff9fd-rdx7g (and 12 other pods): unrestricted capabilities, runAsNonRoot != true, runAsUser=0, seccompProfile
> > Warning: metacontroller-0: seccompProfile
> > Warning: ml-pipeline-9f[4](https://github.com/kubeflow/manifests/actions/runs/10834707411/job/30064531887?pr=2866#step:10:5)fb47b6-r6t4l: allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, runAsUser=0, seccompProfile
> > ```
> 
> I might be wrong but I think the warning are there because those pods have some invalid `securityContextx.capabilities`, I tried locally to apply the `baseline` PSS and looking at the `cache-server` configuration I see:
> 
> ```yaml
>     securityContext:
>       allowPrivilegeEscalation: false
>       capabilities:
>         add:
>         - NET_ADMIN
>         - NET_RAW
>         drop:
>         - ALL
> ```
> 
> And in according to the [documentation](https://kubernetes.io/docs/concepts/security/pod-security-standards/), `NET_ADMIN` and `NET_RAW` are not among the allowed values for `capabilities.add`. I am not an expert so it is just a guess :)
> 
> I did not check when applying the `restricted` one but looking at the warning my guess is that either there is something wrong in the original pod creation or we just need to update the patches
> 
> > We initially suspected that the pods after the patches are applied aren't up or healthy so we added kubectl wait commands for all the deployments of pipelines. Some pods like `metadata-grpc-deployment` failed the timeout.
> 
> This timeout failure might be because the metadata-grpc-deployment could take a looot of time to get ready as it needs to wait for mysql deployment first and then it can start up correctly (it took more than 3 mins in my local env so that might be the reason of the failure - not 100% sure though)

You were right @lampajr , I believe this is what I now see on my local machine. Now to edit the pod I opened it up and saw that exact 11 pods in kubeflow namespace have an istio-init initContainer, which violates the restricted and baseline PSS labels with the securityContext set as you found.

**Comment by juliusvonkohout** [comment]
Like I said, these tests do not make sense for non istio-cni installations. Please do not use the default Istio without cni.

**Comment by biswajit-9776** [comment]
@juliusvonkohout, I changed istio to istio-cni in the installation phase.

**Comment by biswajit-9776** [comment]
Even though the istio-cni uses init containers with proper securityContext attributes, it misses the seccompProfile attribute, that may lead to the following warnings as of now:
```shell
Run ./tests/gh-actions/enable_restricted_PSS.sh
Patching the PSS-restricted labels for namespace istio-system...
namespace/istio-system patched
Patching the PSS-restricted labels for namespace auth...
namespace/auth patched
Patching the PSS-restricted labels for namespace cert-manager...
namespace/cert-manager patched
Patching the PSS-restricted labels for namespace oauth2-proxy...
namespace/oauth2-proxy patched
Patching the PSS-restricted labels for namespace kubeflow...
Warning: existing pods in namespace "kubeflow" violate the new PodSecurity enforce level "restricted:latest"
Warning: cache-server-6c656d7988-v[4](https://github.com/kubeflow/manifests/actions/runs/10868317592/job/30158117073?pr=2866#step:10:5)nzx (and 12 other pods): seccompProfile
Warning: metadata-grpc-deployment-8496ffb98b-wxc6p (and 1 other pod): allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
namespace/kubeflow patched
```

**Comment by juliusvonkohout** [comment]
In the Istio initcontainer configmap or so in the Istio-system namespace you might be able to set the seccomp profile.

**Comment by juliusvonkohout** [comment]
You can also try to set the securitycontext on the pod instead of container level.

**Comment by biswajit-9776** [comment]
The patches targeting deployments with containers, would that be unnecessary? As just the initContainer is failing the seccompProfile...

**Comment by biswajit-9776** [comment]
Hey @juliusvonkohout, are there any patches to templates of configmap in the source code that I could use as a reference? A simple addition of seccompProfile to the sidecar template of the configmap replaces all other data with my patched attribute that I wouldn't want.

**Comment by juliusvonkohout** [comment]
> The patches targeting deployments with containers, would that be unnecessary? As just the initContainer is failing the seccompProfile...

You have to test that.

**Comment by juliusvonkohout** [comment]
> Hey @juliusvonkohout, are there any patches to templates of configmap in the source code that I could use as a reference? A simple addition of seccompProfile to the sidecar template of the configmap replaces all other data with my patched attribute that I wouldn't want.

In the worst case you have to patch it with sed or so.

**Comment by biswajit-9776** [comment]
Okay, so I have used an ugly sed command for the time being @juliusvonkohout. It does solve the istio-containers failing some of the labels.

File: contrib/security/PSS/patches/ml-pipeline-ui.yaml [hunk 2]
```diff
@@ -13,6 +13,14 @@ spec:
           seccompProfile:
             type: RuntimeDefault
           runAsNonRoot: true
+          runAsUser: 1000
+          runAsGroup: 0
           capabilities:
             drop:
             - ALL
+      initContainers:
```
Comment (juliusvonkohout) [inline contrib/security/PSS/patches/ml-pipeline-ui.yaml:21]: Please remove all the istio initcontainer patches from all 
deployment patches. They are injected at runtime and changed by the general istio configuration you modified with sed

File: .github/workflows/pss_test.yaml [hunk 3]
```diff
@@ -41,35 +54,27 @@ jobs:
...
-    - name: Unapply applied baseline values
-      run: |
-        NAMESPACES=("istio-system" "auth" "cert-manager" "oauth2-proxy" "kubeflow")
-        for NAMESPACE in "${NAMESPACES[@]}"; do
-          if kubectl get namespace "$NAMESPACE" >/dev/null 2>&1; then
-            kubectl label namespace $NAMESPACE pod-security.kubernetes.io/enforce-
-          fi
-        done
-        sleep 10
+    # - name: Create dynamic user namespace and check for PSS labels present
```
Comment (juliusvonkohout) [inline .github/workflows/pss_test.yaml:68]: Please remove the debugging parts, such that we can merge soon.

File: .github/workflows/pss_test.yaml [hunk 4]
```diff
@@ -27,12 +27,25 @@ jobs:
...
         kustomize build common/kubeflow-namespace/base | kubectl apply -f -
         ./tests/gh-actions/install_cert_manager.sh
-        ./tests/gh-actions/install_istio_with_ext_auth.sh
-        kustomize build common/istio-1-22/kubeflow-istio-resources/base | kubectl apply -f -
+        ./tests/gh-actions/install_istio-cni.sh
+        kustomize build common/istio-cni-1-22/kubeflow-istio-resources/base | kubectl apply -f -
+
+    - name: Configure istio init container with seccompProfile attribute
+      run: |
+        kubectl get cm istio-sidecar-injector -n istio-system -o yaml > temporary_patch.yaml
```
Comment (juliusvonkohout) [inline .github/workflows/pss_test.yaml:39]: This looks good :-) according to the workflow results it seems to work, but please verify it manually.

**Review by juliusvonkohout — CHANGES_REQUESTED** [review]

File: .github/workflows/pss_test.yaml (hunk 4 above)
Comment (biswajit-9776) [reply .github/workflows/pss_test.yaml:39]: Yes, I did run it manually and opened up the configmap to observe the changes in place as we want. Also, I have removed the comments which I can later add as commits for dynamic namespace.

Comment (biswajit-9776) [reply .github/workflows/pss_test.yaml:39]: The ml-pipeline pod in my local do seem to fail with CrashLoopBackOff as I open it up. Maybe I need to increase my fs.inotify.max_user_{instances, watches} or something.

File: contrib/security/PSS/patches/ml-pipeline-ui.yaml (hunk 2 above)
Comment (biswajit-9776) [reply contrib/security/PSS/patches/ml-pipeline-ui.yaml]: initContainers have been removed in the commit https://github.com/kubeflow/manifests/pull/2866/commits/e40311f4032ed586f8144574bb4e8c742687e5ed

File: .github/workflows/pss_test.yaml (hunk 3 above)
Comment (biswajit-9776) [reply .github/workflows/pss_test.yaml]: Debugging parts along with patch for dynamic namespaces have also been removed in the commit https://github.com/kubeflow/manifests/pull/2866/commits/e40311f4032ed586f8144574bb4e8c742687e5ed. I believe that can be done in a new PR to keep things simple.

**Review by lampajr — COMMENTED** [review]
@biswajit-9776 tests look much better now, the only warning I see at the moment is:
```
Warning: ml-pipeline-bf9f88745-qzmm9: allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
```

**Comment by biswajit-9776** [comment]
> @biswajit-9776 tests look much better now, the only warning I see at the moment is:
> 
> ```
> Warning: ml-pipeline-bf9f88745-qzmm9: allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
> ```

Yes @lampajr, it also looks the same in my local system. Upon opening the pod logs, it doesn't fail the PSS warnings though. It's failing to start up again.

**Comment by juliusvonkohout** [comment]
/lgtm
/approve

lets continue in follow up PRs.

**Pull request merged**